front/src/common/Map/Layers/blankStyle.ts

```diff
@@ -1,24 +1,63 @@
 import { MapProps } from 'react-map-gl/maplibre';
 import { MAIN_API } from 'config/config';
-import { useMemo } from 'react';
+import { useEffect, useMemo, useState } from 'react';
 import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
 
+type Sprite = { url: string; id: string };
+
+const isValidUrl = async (url: string) => {
+  const checkValidity = async (extension: string) => {
+    try {
+      const response = await fetch(`${url}.${extension}`);
+      return response.ok;
+    } catch (error) {
+      return false;
+    }
+  };
+  return (await checkValidity('json')) && checkValidity('png');
+};
+
 export const useMapBlankStyle = (): MapProps['mapStyle'] => {
   const baseURL = MAIN_API.proxy_editoast;
 
   const { data: signalingSystems } =
     osrdEditoastApi.endpoints.getSpritesSignalingSystems.useQuery();
 
+  const getSpriteData = async () => {
+    if (!signalingSystems) return [];
+
+    const ponctualObjectsSprites: Sprite = {
+      url: 'https://static.osm.osrd.fr/sprites/sprites',
+      id: 'default',
+    };
+    const isDefaultSpriteValid = await isValidUrl(ponctualObjectsSprites.url);
+
+    const sprites = await Promise.all([
+      isDefaultSpriteValid ? ponctualObjectsSprites : null,
+      ...signalingSystems.map(async (id) => {
+        const signalingSystemsURL = `${window.location.origin}${baseURL}/sprites/${id}/sprites`;
+        const isValid = await isValidUrl(signalingSystemsURL);
+        return isValid ? { signalingSystemsURL, id } : null;
+      }),
+    ]);
```

Comment:
When you changed your consts (you did right, its clearer now), you missed the fact that the Sprite has an `url` property and you are now passing it as a `signalingSystemsURL` property. To avoid that for next times and increase maintainability, we can do this : 

```suggestion
    const sprites: (Sprite | null)[] = await Promise.all([
      isDefaultSpriteValid ? ponctualObjectsSprites : null,
      ...signalingSystems.map(async (id) => {
        const signalingSystemsURL = `${window.location.origin}${baseURL}/sprites/${id}/sprites`;
        const isValid = await isValidUrl(signalingSystemsURL);
        return isValid ? { url: signalingSystemsURL, id } : null;
      }),
    ]);
```